src/popup-page/modules/app.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useEffect, useContext, useState } from 'react'
+import { getItem, setItem } from '../../utils/storage'
+import { GlobalContext } from '../context/global'
+
+export function App(): JSX.Element {
+  const globalStateContext = useContext(GlobalContext)
+  const { dispatch: changeGlobalState, state: globalState } = globalStateContext
+  const [beeApiUrl, setBeeApiUrl] = useState(globalState.beeApiUrl)
+
+  const asyncInit = async (): Promise<void> => {
+    const storedBeeApiUrl = await getItem('beeApiUrl')
+
+    if (storedBeeApiUrl) {
+      changeGlobalState({ type: 'BEE_API_URL_CHANGE', newValue: storedBeeApiUrl })
+      setBeeApiUrl(storedBeeApiUrl)
+    }
+  }
+
+  useEffect(() => {
+    asyncInit()
+  }, [])
+
+  const handleSubmit = async (event: React.FormEvent<HTMLElement>): Promise<void> => {
+    event.preventDefault()
+    await setItem('beeApiUrl', beeApiUrl)
+    changeGlobalState({ type: 'BEE_API_URL_CHANGE', newValue: beeApiUrl })
+  }
+
+  return (
+    <form id="form-bee-api-url-change" onSubmit={handleSubmit}>
```

Comment:
This form could be extracted in a separate component (for example `SettingsForm`). Then the component state holding the form values and the global state initialization could be well separated.

Reply:
I extracted, but with the name of `BeeApiUrlChangeForm` as it only changes that `beeApiUrl`